docs/topics/wasm/wasm-debugging.md

```diff
@@ -157,15 +157,54 @@ kotlin {
    * ![Step over](wasm-step-over.png){width=30}{type="joined"} Step over to execute the current line and pause on the next line.
    * ![Step out](wasm-step-out.png){width=30}{type="joined"} Step out to execute the code until it exits the current function.
 
-![Debug controls](wasm-debug-controls.png){width=700}
+   ![Debug controls](wasm-debug-controls.png){width=700}
 
 6. Check the **Call stack** and **Scope** panes to trace the sequence of function calls and pinpoint the location of any errors.
 
-![Check call stack](wasm-debug-scope.png){width=700}
+   ![Check call stack](wasm-debug-scope.png){width=700}
+
+   For an improved visualization of the variable values, see [Enhance your debugging with custom formatters](#enhance-your-debugging-with-custom-formatters).
 
 7. Make changes to your code and [run the application](#run-the-application) again to verify that everything works as expected.
 8. Click on the line numbers with breakpoints to remove the breakpoints.
 
+### Enhance your debugging with custom formatters
+
+In addition to the default debugging experience, you can set up custom formatters to display and locate variable values in a more user-friendly and comprehensible manner.
+
+![Kotlin/Wasm improved debugger](wasm-debugger-improved.png){width=700}
+
+This implementation is supported across major browsers like Firefox and Chromium-based as
+it uses the [custom formatters API](https://firefox-source-docs.mozilla.org/devtools-user/custom_formatters/index.html).
+
+To set up custom formatters for an improved debugging experience:
+
+1. Add the following compiler argument to the `wasmJs` compiler options:
+
+```kotlin
+kotlin {
+    wasmJs {
+        …
+        compilerOptions {
+            freeCompilerArgs.add("-Xwasm-debugger-custom-formatters")
+        }
+    }
+}   
+```
+
+2. Enable the **Custom formatters** feature in your browser.
+
+   In the Chrome DevTools, it’s placed in **Settings | Preferences | Console**:
```

Comment:
This could also be a list, for example:

2. Enable the **Custom formatters** feature in your browser:
    * In the Chrome DevTools,
    * In the Firefox Developer Tools,